### PR TITLE
Rec: Implement computing results of recognizer

### PR DESCRIPTION
**Pull request by KnapSac**

### CODE REVIEW

File: PatternPal/PatternPal.Core/CheckResult.cs [hunk 1]
```diff
@@ -70,4 +95,15 @@ internal class NodeCheckResult : ICheckResult
...
+    /// <inheritdoc />
+    public required int DependencyCount { get; init; }
+
+    /// <inheritdoc />
+    public required INode ? MatchedNode { get; init; }
+
+    /// <summary>
+    /// All sub-<see cref="ICheckResult"/>s are the result of a <see cref="NodeCheck{TNode}"/>.
+    /// </summary>
+    public bool NodeCheckCollectionWrapper { get; set; }
```
Comment (LvanWerven) [inline PatternPal/PatternPal.Core/CheckResult.cs:108]: It is not quite clear for me what this is or how it is evaluated, could you elaborate on that?

File: PatternPal/PatternPal/Services/RecognizerService.cs
Comment (LvanWerven) [inline PatternPal/PatternPal/Services/RecognizerService.cs:1]: Check the Eslint warnings, think about boyscout!

File: PatternPal/PatternPal.Core/Checks/NotCheck.cs
[the remark this answers is not in the document]
Comment (KnapSac) [reply PatternPal/PatternPal.Core/Checks/NotCheck.cs:1]: I don't know exactly what Rutger is trying to test, but checking whether a `NotCheck` is correct, is handled by `RecognizerRunner.FilterResults`. And the result type of a `NotCheck` is `NotCheckResult`, so he should definitely be able to tell whether he's looking at the result of a `NotCheck` or not.

**Review by LvanWerven — APPROVED** [review]
Looking goodd

**Review by LvanWerven — APPROVED** [review]
Looking goodd

**Pull request merged**